### PR TITLE
Update container and basic text doc.

### PR DESCRIPTION
**Pull request by sylane**

### CODE REVIEW

File: source/index.html.md [hunk 1]
```diff
@@ -1011,11 +1011,38 @@ label | string | yes | a label for the checkbox, positioned directly to the righ
 
 ## Container component
 
-Generic root container for its `children`, it does not change the UI itself.
+Component that could be a root component or a child of any root components, other container components or [ListItem](#list-item-component) components.
+The allowed children depends on the parent components.
```
Comment (PaulLeCam) [inline source/index.html.md]: Couldn't we do the validation based on the parent + root rather than parent + grandparent as we discussed?

**Comment by howleysv** [comment]
Is this still relevant?

**Comment by PaulLeCam** [comment]
It's a bit outdated with the other PRs that got merged. I think it's good to document the validation but I find these explanations quite confusing.
What is the current server implementation for the validation? Does it still do it on parent + grandparent or parent + root?

**Comment by sylane** [comment]
It is more or less parent + root... Not exactly like that because it is more parent + "is in form" + context but the result is similar.

**Comment by PaulLeCam** [comment]
Seems to me this is making things more complex to understand, what is the problem of having each containing component describe itself what are the supported `children`?
I updated the [components section](http://developer.mainframe.com/#components) a bit over the last few PRs, so the supported root components are more documented, could that be enough?

**Comment by sylane** [comment]
How it is working internally is not relevant if it supports the doc. I don't care much if it get merged or not, less conflict resolution to deal with. Someone just tell me if it is relevant or if it should be dropped.

**Comment by howleysv** [comment]
@PaulLeCam I think the issue is around documenting the behaviour when `Container` is used as a non-root component.

**Comment by PaulLeCam** [comment]
I think we should present it the other way around, no? `Container` only cares about its children, whether it is root or not doesn't change what it supports.
I'd rather document that when the `Message` root is used, some components can't be used in the tree, what do you think?

**Comment by howleysv** [comment]
Sounds fine to me. @sylane ?

**Comment by sylane** [comment]
We would be faster adding what you think is missing to the doc than asking me to try to update this PR. This PR would change completely anyway. If you want me to do a new PR anyway we would have to clear up what you want exactly. Container list all possible children (as it is in master) and in the Form root component we state that it cannot contain any of the XXX components in its component tree, and in the Message root component the same ? Note that container being a root component too it would have to have the list of components that cannot be in its sub-tree...

**Comment by PaulLeCam** [comment]
I added more info about the Message/Container exceptions in https://github.com/thusfresh/bots-documentation/pull/27 please have a look!

**Pull request closed without merging**